app/controllers/volunteers_controller.rb

```diff
@@ -129,14 +121,11 @@ def create
     return unless check_permissions(@volunteer)
     # can't set admin bits from CRUD controls
     @volunteer.admin = false
-    @volunteer.assignments.each{ |assingment| assingment.admin = false }
+    @volunteer.assignments.each { |assignment| assignment.admin = false }
+
     if @volunteer.save
       flash[:notice] = 'Created successfully.'
-      unless session[:my_return_to].nil?
-        redirect_to(session[:my_return_to])
-      else
-        index
-      end
+      redirect_to(session[:my_return_to] || index)
     else
       flash[:error] = "Didn't save successfully :(. #{@volunteer.errors.full_messages.to_sentence}"
```

Comment:
This is the end of the codepath, so redirect_to is in the if and then in the else is the render new. 

This is also a call to the index method in the `|| index` not a redirect to index.

I had to actually fix this to be this after I merged to master as it failed a test:

`session[:my_return_to].present? ? redirect_to(session[:my_return_to]) : index `

Does that make sense? Its not super straightforward, but you need an explicit return if there is other code in the block that could still be executed.